eradiate/scenes/biosphere/tests/test_abstract_tree.py

```diff
@@ -0,0 +1,211 @@
+                "n_leaves": 100,
+                "l_horizontal": 10.0,
+                "l_vertical": 1.0,
+                "leaf_radius": 10.0,
+                "leaf_radius_units": "cm",
+            },
+            "trunk_height": 2.0,
+            "trunk_radius": 0.2,
+            "trunk_reflectance": 0.5,
+        }
+    )
+
+    tree1 = AbstractTree.from_dict(
+        {
+            "leaf_cloud": {
+                "construct": "cuboid",
+                "n_leaves": 100,
+                "l_horizontal": 10.0,
+                "l_vertical": 1.0,
+                "leaf_radius": 10.0,
+                "leaf_radius_units": "cm",
+            },
+            "trunk_height": 2.0,
+            "trunk_radius": 0.2,
+            "trunk_reflectance": 0.5,
+        }
+    )
+
+    tree2 = AbstractTree(
+        leaf_cloud=LeafCloud.from_dict(
+            {
+                "construct": "cuboid",
+                "n_leaves": 100,
+                "l_horizontal": 10.0,
+                "l_vertical": 1.0,
+                "leaf_radius": 10.0,
+                "leaf_radius_units": "cm",
+            }
+        ),
+        trunk_height=2.0,
+        trunk_radius=0.2,
+        trunk_reflectance=0.5,
+    )
+
+    assert tree1.trunk_height == tree2.trunk_height
+    assert tree1.trunk_radius == tree2.trunk_radius
+    assert tree1.trunk_reflectance == tree2.trunk_reflectance
+
+    assert np.allclose(tree1.leaf_cloud.leaf_positions, tree2.leaf_cloud.leaf_positions)
+    assert np.allclose(
+        tree1.leaf_cloud.leaf_orientations, tree2.leaf_cloud.leaf_orientations
+    )
+    assert np.all(
+        tree1.leaf_cloud.leaf_transmittance == tree2.leaf_cloud.leaf_transmittance
+    )
+    assert np.all(
+        tree1.leaf_cloud.leaf_reflectance == tree2.leaf_cloud.leaf_reflectance
+    )
+
+
+def test_abstract_tree_kernel_dict(mode_mono):
+    ctx = KernelDictContext()
+
+    """Partial unit testing for :meth:`LeafCloud.kernel_dict`."""
+    tree_id = "my_tree"
+    cloud_id = "my_cloud"
+    tree = AbstractTree(
+        leaf_cloud=LeafCloud(
+            id=cloud_id,
+            leaf_positions=[[0, 0, 0], [1, 1, 1]],
+            leaf_orientations=[[1, 0, 0], [0, 1, 0]],
+            leaf_radii=[0.1, 0.1],
+            leaf_reflectance=0.5,
+            leaf_transmittance=0.5,
+        ),
+        id=tree_id,
+        trunk_height=2.0,
+        trunk_radius=0.2,
+        trunk_reflectance=0.5,
+    )
+
+    kernel_dict = tree.kernel_dict(ctx=ctx)
+
+    # The BSDF is bilambertian with the parameters we initially set
+    assert kernel_dict[f"bsdf_{cloud_id}"] == {
+        "type": "bilambertian",
+        "reflectance": {"type": "uniform", "value": 0.5},
+        "transmittance": {"type": "uniform", "value": 0.5},
+    }
+
+    # Leaves are disks
+    for shape_key in [f"{cloud_id}_leaf_0", f"{cloud_id}_leaf_1"]:
+        assert kernel_dict[shape_key]["type"] == "disk"
+
+    # Kernel dict is valid
+    assert KernelDict.new(kernel_dict).load()
```

Comment:
This is a duplicate of `test_biosphere_tree_abstract.py`: it can be removed safely.